src/utilities/idf/WorkspaceObject.cpp

```diff
@@ -431,8 +431,11 @@ namespace detail {
     }
 
     // regular field -- name or data
-    if ((index == 0) && (iddObject().hasNameField())) {
-      return setName(value, checkValidity).has_value();
+    if (iddObject().hasNameField()) {
+      boost::optional<unsigned> nameIndex = iddObject().nameFieldIndex(); 
+      if (nameIndex && (*nameIndex == index)) {
+        return setName(value, checkValidity).has_value();
+      }
```

Comment:
```suggestion
   // regular field -- name or data
    if ((iddObject().hasNameField()) && (index == iddObject().nameFieldIndex().get())) {
      return setName(value, checkValidity).has_value();
    }  // name
```

if hasNameField, I don't think you also need to check if nameIndex exits.